File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nMap = {}

        for i in range (0, len(nums)):
            if target - nums[i] in nMap:
                return [nMap[target-nums[i]], i]
            nMap[nums[i]] = i
        
        return [-1, -1]

File: Data Structures & Algorithms/remove-element/submission-3.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        // 2, 2, 3, 3
        //   ik 
        int i = 0;
        int k = nums.size()-1;
        while(i <= k)
        {
            if(nums[i] == val)
            {
                swap(nums[i], nums[k]);
                k--;
            }
            else
                i++;
        }
        return k+1;
    }
};